com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -504,6 +514,252 @@ internal void InvokeBehaviourOnGainedOwnership()
             }
         }
 
+        internal void InvokeBehaviourOnNetworkObjectParentChanged(NetworkObject parentNetworkObject)
+        {
+            for (int i = 0; i < ChildNetworkBehaviours.Count; i++)
+            {
+                ChildNetworkBehaviours[i].OnNetworkObjectParentChanged(parentNetworkObject);
+            }
+        }
+
+        private bool m_IsReparented;
+        private ulong? m_LatestParent;
+        private Transform m_CachedParent;
+
+        internal void SetCachedParent(Transform parentTransform)
+        {
+            m_CachedParent = parentTransform;
+        }
+
+        internal static void WriteNetworkParenting(NetworkWriter writer, bool isReparented, ulong? latestParent)
+        {
+            writer.WriteBool(isReparented);
+            if (isReparented)
+            {
+                var isLatestParentSet = latestParent != null && latestParent.HasValue;
+                writer.WriteBool(isLatestParentSet);
+                if (isLatestParentSet)
+                {
+                    writer.WriteUInt64Packed(latestParent.Value);
+                }
+            }
+        }
+
+        internal static (bool IsReparented, ulong? LatestParent) ReadNetworkParenting(NetworkReader reader)
+        {
+            ulong? latestParent = null;
+            bool isReparented = reader.ReadBool();
+            if (isReparented)
+            {
+                var isLatestParentSet = reader.ReadBool();
+                if (isLatestParentSet)
+                {
+                    latestParent = reader.ReadUInt64Packed();
+                }
+            }
+
+            return (isReparented, latestParent);
+        }
+
+        internal (bool IsReparented, ulong? LatestParent) GetNetworkParenting() => (m_IsReparented, m_LatestParent);
+
+        internal void SetNetworkParenting(bool isReparented, ulong? latestParent)
+        {
+            m_IsReparented = isReparented;
+            m_LatestParent = latestParent;
+        }
+
+        public bool TrySetParent(Transform parent, bool worldPositionStays = true)
+        {
+            return TrySetParent(parent.GetComponent<NetworkObject>(), worldPositionStays);
+        }
+
+        public bool TrySetParent(GameObject parent, bool worldPositionStays = true)
+        {
+            return TrySetParent(parent.GetComponent<NetworkObject>(), worldPositionStays);
+        }
+
+        public bool TrySetParent(NetworkObject parent, bool worldPositionStays = true)
```

Comment:
So, we want to silently fail in these cases or what about throwing an exception?

Reply:
I don't think returning a `bool` back from a method called `TryXXX` is silencing fail cases. it is pretty much like .NET `Dictionary.TryAdd()` API.